crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -65,7 +65,70 @@ impl Solution {
         weth: eth::WethAddress,
         gas: Option<eth::Gas>,
         fee_handler: FeeHandler,
+        surplus_capturing_jit_order_owners: &HashSet<eth::Address>,
     ) -> Result<Self, error::Solution> {
+        tracing::error!(?surplus_capturing_jit_order_owners);
+        // Surplus capturing JIT orders behave like Fulfillment orders. They capture
+        // surplus, pay network fees and contribute to score of a solution.
+        // To make sure that all the same logic and checks get applied we convert them
+        // right away.
+        for trade in &mut trades {
```

Comment:
I guess once merge for a cleanup we can delete the domain "Trade" as it is and  keep it just a "Fulfillment", right? 🤔